src/peft/tuners/lora/layer.py

```diff
@@ -512,11 +560,15 @@ def forward(self, x: torch.Tensor, *args: Any, **kwargs: Any) -> torch.Tensor:
                 scaling = self.scaling[active_adapter]
                 x = x.to(lora_A.weight.dtype)
 
-                if not self.use_dora[active_adapter]:
-                    result = result + lora_B(lora_A(dropout(x))) * scaling
-                else:
+                if self.use_dora[active_adapter]:
                     x = dropout(x)
                     result = result + self._apply_dora(x, lora_A, lora_B, scaling, active_adapter)
+                elif self.use_wlora[active_adapter]:
+                    wlora_scale = self._cal_wlora_scale(active_adapter)
```

Comment:
I think I understand your suggestion. So there are two ways we can combine dlora with wlora:

### wlora and dlora working separately

So in this case (this is the case that we already support in the PR), we can have let's say three Lora modules, e.g., lora_1, lora_2, and lora_3. We can set lora_1 as dlora and lora_2 and lora_3 as wlora. So the forward formulation looks like this:
```python
result = result
        + self._apply_dora(x, lora_1_A, lora_1_B, scaling_1, lora_1)
        + lora_2_B(lora_2_A(dropout_2(x))) * scaling_2 * wlora_scale_2
        + lora_3_B(lora_3_A(dropout_3(x))) * scaling_3 * wlora_scale_3
```
where `wlora_scale_2 + wlora_scale_3 = 1`

### wlora and dlora working togther
In this case (that we do not currently support and I it helps dlora at all) we use two LoRAs (e.g. lora_1 and lora_2) inside dlora and we need to change _apply_dora function to accept wlora parameters like this:
```python
result = result
        + self._apply_dora_and_wlora(x, lora_1_A, lora_1_B, scaling_1, wlora_scale_1, lora_2_A, lora_2_B, scaling_2, wlora_scale_2,)
```

Overall, I think with the current implementation, wlora works with other modules and we don't have to worry about conflicts since it can be trained and combined with other PEFT methods.

Reply:
Not quite sure if I understand your suggestion completely. Probably you mean if we "unroll" the for loop for the different adapters inside of `forward`, this is what it would look like?

Would it not work if we just apply the wlora scale to the existing `scaling` argument and the rest takes care of itself?